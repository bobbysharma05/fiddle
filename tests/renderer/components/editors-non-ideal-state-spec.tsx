import { mount } from 'enzyme';
import { vi } from 'vitest';

import { renderNonIdealState } from '../../../src/renderer/components/editors-non-ideal-state';
import { EditorMosaic } from '../../../src/renderer/editor-mosaic';

describe('renderNonIdealState()', () => {
  let editorMosaic: EditorMosaic;

  beforeEach(() => {
    ({ editorMosaic } = window.app.state);
  });

  it('renders a non-ideal state', () => {
    expect(renderNonIdealState({} as EditorMosaic)).toMatchSnapshot();
  });

  it('handles a click', () => {
    const resetLayoutSpy = vi.spyOn(editorMosaic, 'resetLayout');
    const wrapper = mount(renderNonIdealState(editorMosaic));
    wrapper.find('button').simulate('click');
    expect(resetLayoutSpy).toHaveBeenCalledTimes(1);
  });
});
