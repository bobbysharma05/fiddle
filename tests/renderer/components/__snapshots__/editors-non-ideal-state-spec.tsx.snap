// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderNonIdealState() > renders a non-ideal state 1`] = `
<Blueprint3.NonIdealState
  action={
    <Blueprint3.Button
      onClick={[Function]}
      text="Reset editors"
    />
  }
  description="You have closed all editors. You can open them again with the button below or in the sidebar menu!"
  icon="applications"
/>
`;
